### PR TITLE
Telemetry collection

### PR DESCRIPTION
**Pull request by generall**
Enable optional telemetry and panics reporting

### CODE REVIEW

File: src/startup.rs [hunk 1]
```diff
@@ -28,19 +30,24 @@ pub fn setup_logger(log_level: &str) {
...
-            log::error!("Panic occurred{loc}: {s}");
+            s
         } else {
-            log::error!("Panic occurred{loc}. Payload not captured as it is not a string.");
+            "Payload not captured as it is not a string."
+        };
+        log::error!("Panic occurred{loc}: {message}");
+
+        if reporting_enabled {
+            ErrorReporter::report(message, &reporting_id, Some(&loc));
```
Comment (agourlay) [inline src/startup.rs:50]: Were you able to manually test that the `ErrorReporter` did the job during a panic?

**Review by agourlay — APPROVED** [review]
Good for me if it was tested manually :+1:

File: src/common/error_reporting.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        let client = reqwest::blocking::Client::new();
+
+        let report = serde_json::json!({
+            "id": reporting_id,
+            "error": error,
+            "backtrace": backtrace.unwrap_or(""),
+        });
+
+        let data = serde_json::to_string(&report).unwrap();
+        let _resp = client
```
Comment (agourlay) [inline src/common/error_reporting.rs:22]: Should we maybe log failures?

File: src/startup.rs (hunk 1 above)
Comment (generall) [reply src/startup.rs:50]: nope

File: src/common/error_reporting.rs (hunk 2 above)
Comment (generall) [reply src/common/error_reporting.rs:22]: no, reporting in 100% optional, in some cases it is expected not to work (like no internet env, for example). So we won't log anything here

File: src/startup.rs (hunk 1 above)
Comment (agourlay) [reply src/startup.rs:50]: I have tested manually, it works :+1:

**Pull request merged**